### PR TITLE
feat(build-editor): lock armor weight per set + dedupe gear picker variants

### PR DESCRIPTION
**Pull request by BraydenPB**
## What & why

Two confusing behaviors in the build editor's gear picker:

1. **The weight chip cycled for everything.** You could click any apparel weight chip to cycle Light → Medium → Heavy — even for gear that's locked to one weight in-game (a mythic like Huntsman's Warmask is Medium-only; Gaze of Sithis is Heavy-only; most overland/dungeon/trial drops are single-weight).
2. **"Zaan shoulders has 24 results."** The raw item data carries one entry per quality/level/trait variant, so a single monster set produced ~24 identical-looking rows.

## Weight locking

- Sets that only exist in one armor weight now show a **read-only weight badge with a lock icon** instead of a cyclable chip. Crafted sets (Hunding's, Julianos) and monster sets — which genuinely come in all three weights — keep the cyclable control.
- **Source of truth:** the **LibSets `LIBSETS_TABLEKEY_SETS_ARMOR_TYPES` table** (the same addon data the project already uses), keyed `[1]=light [2]=medium [3]=heavy → setIds`. Generated into `setArmorWeights.generated.ts` by `scripts/generate-set-armor-weights.mjs`.
  - The local `eso-globals-item-set-collections.json` `weight` field is a *player-collection dump* and is incomplete (shows Mother's Sorrow as light+medium only), so it is **not** used.
  - Verified against authoritative web sources (eso-hub, hacktheminotaur, arzyelarmory): **19/23 sampled sets matched exactly at high confidence**. The 4 "mismatches" were LibSets' rare two-weight entries — all verified to actually be single-weight (legacy items / one mis-flagged slot), corrected via explicit overrides in the generator. The result also agrees with `armorUtils.ts` on all 12 mythics.
- Selecting a locked set **persists its correct weight** (and preserves the slot's existing trait/enchant). Picker rows show an inline weight badge so the locked weight is visible before equipping.

## Variant dedupe

- New `getCanonicalItemsBySlot()` collapses each set to a single canonical item per slot (lowest item id), removing the quality/level/trait duplicates. **Zaan shoulders: 24 → 1.** The build editor never distinguished those variants (weight/trait/enchant are separate chips). `getItemsBySlot` is left untouched for the legacy item picker.
- Picker set rows are now **directly selectable** (click to equip) with a separate toggle to peek set bonuses. The EQUIPPED indicator matches by **set name**, so a previously-equipped non-canonical variant still highlights.

## Verification

Verified live in the running app:
- Zaan shoulders search → **1 result** (was 24)
- Mother's Sorrow on Head → **locked Light badge** (clicks do nothing)
- Zaan on Shoulders (monster set) → **free, cyclable Heavy chip**

Tests:
- New `setArmorWeights.test.ts` (35 cases: mythics, single-weight drops, free crafted/monster sets, two-weight overrides, edge cases) + canonical-dedupe coverage in `itemSlotValidator.test.ts`.
- Full relevant suite passes (3164 tests); `tsc` + `eslint` clean on changed files.

## Regenerating the data

```
gh api repos/Baertram/LibSets/contents/LibSets/Data/LibSets_Data_SetOtherData.lua?ref=LibSets-reworked --jq '.content' | base64 -d > .scratch/libsets_otherdata.lua
node scripts/generate-set-armor-weights.mjs
```

### CODE REVIEW

File: src/features/build-editor/components/pickers/GearPicker.tsx [hunk 1]
```diff
@@ -100,7 +151,9 @@ function getSetGroupsForSlot(targetSlot: SlotType): SetGroupResult {
 
 function buildSetGroups(targetSlot: SlotType): SetGroupResult {
   const setSummaries = getAvailableSetsForSlot(targetSlot);
-  const allItems = getItemsBySlot(targetSlot);
+  // One canonical item per set+slot — collapses the ~24 level/quality/trait
+  // variants the raw data carries (e.g. "Zaan Shoulders") down to a single row.
+  const allItems = getCanonicalItemsBySlot(targetSlot);
```
[the remark this answers is not in the document]
Comment (BraydenPB) [reply src/features/build-editor/components/pickers/GearPicker.tsx]: Resolved. Weapon variants are preserved: `getCanonicalItemsBySlot` now takes a slot-aware key (`cc3432fc`) — apparel/jewelry dedupe by set name, but weapon/offhand key by the per-item weapon **type** (incl. staff element), so each bow/staff/greatsword/etc. survives as its own selectable row. The browse row expands to its types and search matches the derived type name. Robustness for lazy/stale icon data (`isWeaponTypeResolved`, non-collapsing per-item fallback, non-selectable until resolved) landed across `1b5414c8`→`a12e8691`. Covered by tests in `itemSlotValidator.test.ts` (weapon-aware key) and `GearPicker.{iconRace,iconError,staleIcon,mixedWeapons}.test.tsx`.

File: src/features/build-editor/components/pickers/EquipmentPicker.tsx [hunk 2]
```diff
@@ -89,7 +95,8 @@ const SlotRowComponent: React.FC<SlotRowProps> = ({
       setName={info?.setName ?? null}
       isDisabled={Boolean(disabledReason)}
       disabledReason={disabledReason}
-      weight={piece?.weight}
+      weight={lockedWeight ?? piece?.weight}
+      lockedWeight={lockedWeight}
```
[the remark this answers is not in the document]
Comment (BraydenPB) [reply src/features/build-editor/components/pickers/EquipmentPicker.tsx:137]: Resolved. Locked weight is now authoritative in calculations, not just display. `countArmorWeights` resolves via the shared `resolveApparelWeight(id, storedWeight)` (`cc3432fc`, refactored to a shared helper in `a64913e6`) — the in-game locked weight wins over an absent/stale `piece.weight`, so imported/CSPS apparel counts correctly. The weight is also serialized into CSPS `type` on export (SavedVariables + export-code + native-code) and deserialized on import, giving a full round-trip (`a64913e6`/`7097a555`/`ae79162f`), and gated to apparel only so it can't leak onto a locked set's jewelry/weapon. Covered by `countArmorWeights.test.ts`, `buildReplacementPiece.test.ts`, and CSPS export/import round-trip tests.
